### PR TITLE
libsingular_julia: id_prune_map_v

### PR DESCRIPTION
**Pull request by hannes14**

### CODE REVIEW

File: deps/src/ideals.cpp [hunk 1]
```diff
@@ -404,7 +393,6 @@ void singular_define_ideals(jlcxx::Module & Singular)
 
   Singular.method("id_res", &id_res_helper);
   Singular.method("id_mres_map", &id_mres_map_helper);
-  Singular.method("id_prune_map", &id_prune_map_helper);
```
Comment (fingolfin) [inline deps/src/ideals.cpp:407]: You remove `id_prune_map` but there is a call to it in `src/module/module.jl` line 405

Comment (fingolfin) [reply deps/src/ideals.cpp:407]: Specifically in the implementation of the function `prune_with_map`, which also has a test case.

Comment (hannes14) [reply deps/src/ideals.cpp:407]: re-added id_prune_map_helper

File: deps/src/ideals.cpp [hunk 2]
```diff
@@ -405,6 +416,7 @@ void singular_define_ideals(jlcxx::Module & Singular)
   Singular.method("id_res", &id_res_helper);
   Singular.method("id_mres_map", &id_mres_map_helper);
   Singular.method("id_prune_map", &id_prune_map_helper);
+  Singular.method("id_prune_map_v", &id_prune_map_v_helper);
```
Comment (fingolfin) [inline deps/src/ideals.cpp:419]: But what does this do and which code is going to call it?

Comment (hannes14) [reply deps/src/ideals.cpp:419]: id_prune_map and id_prune_map_v are 2 interfaces to computing a minimal embedding of the given module (in Singular). Return the minimal embedding, the map, and (in the 2nd case) an array how to map indices.
Should be documented and tested by the Singular.jl routines prune_map   and (coming) prune_map_v

**Review by wdecker — APPROVED** [review]

**Review by fingolfin — APPROVED** [review]

**Pull request merged**